crates/typst-cli/src/package.rs

```diff
@@ -1,14 +1,55 @@
 use std::fs;
 use std::io::{self, Write};
 use std::path::{Path, PathBuf};
+use std::sync::Arc;
 
 use codespan_reporting::term::{self, termcolor};
+use once_cell::sync::Lazy;
 use termcolor::WriteColor;
 use typst::diag::{PackageError, PackageResult};
 use typst::syntax::PackageSpec;
 
 use super::color_stream;
 
+/// HTTP request agent.
+static AGENT: Lazy<ureq::Agent> = Lazy::new(|| {
+    // Get the network proxy config from the environment.
+    let proxy = env_proxy::for_url_str("https://typst.org")
```

Comment:
Some requests will happen to `typst.org`, but others to `github.com`. This will probably be a problem here.

Reply:
Hmm, yeah, I don't think it's efficient to have a different agent for each URL we might need to access, and `ureq` doesn't seem to have a great API for allowing you to disable the proxy for certain domains.

The only reason we provide the URL here is so that the `env_proxy` check can make sure the domain isn't in the `no_proxy` environment variable. I'm wondering if _maybe_ it'd be easiest to just ignore the `no_proxy` environment variable and just use the proxy, if set, for all requests?

It's not proper, but the only alternative I can think of is to have two agents and create a wrapper around the `get` method that would check whether the URL is in the `no_proxy` list and use the non-proxied agent if it is, otherwise it just uses the proxied agent.

I feel like ignoring `no_proxy` won't be a problem for most people, I'm not sure if it's worth the implementation complexity.

Reply:
Let's just ignore it then

Reply:
I would ignore the `no_proxy` argument and never put any default values for `no_proxy`.

Reply:
I got a little swamped, so if anybody wants to come in and finish this up feel free. Otherwise, also feel free to ping me again on this if I don't get to it within a week or so.

Reply:
> I got a little swamped, so if anybody wants to come in and finish this up feel free. Otherwise, also feel free to ping me again on this if I don't get to it within a week or so.

I feel you, I have a gradient PR in the works (although I haven't opened the PR yet) and I have been swamped with exams, can't wait to get back to it 😄